Source/UI/NewUI/BillingForm/View/Error/BillingFormTextFieldErrorView.swift

```diff
@@ -0,0 +1,68 @@
+import UIKit
+
+final class BillingFormTextFieldErrorView: UIView {
+    private var style: ErrorInputLabelStyle = DefaultErrorInputLabelStyle()
+    
+    lazy var headerLabel: UILabel = {
+        let view = UILabel()
+        view.text = style.text
+        view.font = style.font
+        view.textColor = style.textColor
+        view.textAlignment = .justified
+        view.numberOfLines = 0
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = .white
+        return view
+    }()
+    
+    lazy var image: UIImageView = {
+        let view = UIImageView()
+        view.image = "warning".vectorPDFImage(forClass: BillingFormTextFieldErrorView.self)
+        view.tintColor = style.tintColor
+        view.contentMode = .scaleAspectFit
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.backgroundColor = .white
+        return view
+    }()
+    
+    init(style: ErrorInputLabelStyle) {
+        super.init(frame: .zero)
+        self.style = style
+        setupViews()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension BillingFormTextFieldErrorView {
+
+    private func setupViews(){
+        backgroundColor = style.backgroundColor
+        setupHeaderLabel()
+        setupImageView()
```

Comment:
Question : @alex-ioja-yang-cko any idea why it would crash if kept in separate function ? wouldn't the compiler treat that as same. My understanding was the different functions is really there for readability. I am keep to know if something crashes and we can avoid that.

Reply:
It doesn't crash if kept in separate functions, but having separate functions can make it appear simpler than it is and hide the complex dependency on the layout side.

If you create a constraint between 2 views that don't belong to the same stack, the application will crash. So keeping the correct layout as it was designed is very important. In this case, abstracting that complexity behind easy to navigate functions may give a false confidence that the code is easy. Of course, this does improve developer experience and can be kept, but we do need to be very careful on the code being committed.

Imagine tomorrow a new dev comes and copies this approach to a new view, see it running and all good, do some method shuffling as part as code review (lets say swap order on those 2 so the trailing edge faces a certain way, or order alphabetically 😜). If there is no test to crash when layout is created, that can sit silently for a while.
Take it further, a test may exist, but doing the setup view on init may be using resources too eagerly. You may later optimise by moving the setup to a lifecycle method that increases likelihood of reaching display (like `willMove(toSuperview`). Most unit tests will not trigger that method, allowing you to _safely refactor the order of the methods_ and have it start crashing. Process can obviously mitigate, but even UITests may not 100% cover all elements.

Reply:
oversimplified example for crash on constraint creation:
```
let superview = UIView()
let subview1 = UIView()
let subview2 = UIView()

superview.addSubview(subview1)
subview1.pinTrailingEdge(to: subview2) // This crashes as subview 1 and subview 2 don't have the same hierarchy
```

Reply:
I love keeping things simple, and I'd always suggest we break code down so Patrick can come and read the code like a story. But Layout Constraints are 1 thing I wouldn't want to oversimplify and I'd say keeping it harder encourages dev to really look at what they're doing 👀

Reply:
Renaming method to `setupViewsInOrder`